developer_notes.md

```diff
@@ -0,0 +1,22 @@
+Developer notes
+===============
+
+Summary of changes in regards to the migration to v2
+----------------------------------------------------
+
+  * Switched to a `setup.cfg`/`pyproject.toml` based installation mechanism
+  * Switched to a `src/` based package layout
+  * Exposed main CLI command as a proper entry point
+    * Separated `gmx2qmmm.py` into argument parsing (`cli.py`) and job execution (`app.py`) part
+    * Removed `assess_job` function that did nothing
```

Comment:
I will write that function as soon as possible. It will check the input for correctness and add default values or errors where input is missing.